### ISSUE TITLE
Make TCP error panic configurable

### ISSUE THREAD

**Issue by rzezeski** [oxidecomputer/opte#194]
In development it has been useful to have OPTE panic anytime it hits an unexpected transition in a TCP flow. However, with Omicron development ramping up and OPTE being used more and more, it is a bit counterproductive to crash people's machines when they are trying to work on other aspects of the system. A good middle ground is to allow the behavior to be configurable; by default no panic occurs, but in cases where panicking would be useful it can be turned on. Typically something like this is done at compile-time, like the illumos `ASSERT` family of macros. However, in this case I want to make it dynamic (via a global variable). Also, by default, when the panic behavior is not enabled, an SDT probe should be fired in it's place, so that one can use DTrace to catch these dynamically in a production context.

While this issue is specifically about the current `Port::bad_packet_err()`, this dynamic behavior applies to any site where it might be advantageous to panic in order to facilitate better debugging in development. And all sites should share the same global to control this behavior. Those sites will not be changed as part of this work, but I wanted to note that this should be the direction going forward.

**Comment by rzezeski** [comment]
To test this I had to temporarily change the `tcp_state.rs` code to **NOT** handle outgoing RSTs and then run through the scenario described in #179.

```
rpz@kalm:~/opte-dtrace$ pfexec dtrace -qn 'tcp-err { printf("%s %s %s\n", stringof(arg1), stringof(arg0), stringof(arg4)); }'
opte14 out unexpected TCP segment dir: OUT, flow: TCP:172.30.0.5:22:10.0.0.10:49547, state: LISTEN, flags: 0x4
opte14 in unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:10.0.0.10:49547, state: LISTEN, flags: 0x18
opte14 in unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:10.0.0.10:49547, state: LISTEN, flags: 0x18
opte14 in unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:10.0.0.10:49547, state: LISTEN, flags: 0x18
opte14 in unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:10.0.0.10:49547, state: LISTEN, flags: 0x18
opte14 in unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:10.0.0.10:49547, state: LISTEN, flags: 0x18
opte14 in unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:10.0.0.10:49547, state: LISTEN, flags: 0x18
opte14 in unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:10.0.0.10:49547, state: LISTEN, flags: 0x18
opte14 in unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:10.0.0.10:49547, state: LISTEN, flags: 0x18
opte14 in unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:10.0.0.10:49547, state: LISTEN, flags: 0x18
```

**Comment by rzezeski** [comment]
Addressed in dfe4a550642005ded29ff7271bcf96a457529f21.
